spec/features/flights/index_spec.rb

```diff
@@ -0,0 +1,237 @@
+        expect(page).to_not have_content(@passenger3.name)
+        expect(page).to_not have_content(@passenger4.name)
+        expect(page).to_not have_content(@passenger5.name)
+        expect(page).to_not have_content(@passenger7.name)
+      end
+
+      within("#flight-#{@flight4.id}") do
+        expect(page).to have_content("Flight Airline: #{@airline2.name}")
+        expect(page).to have_content(@passenger4.name)
+        expect(page).to have_content(@passenger5.name)
+        expect(page).to have_content(@passenger6.name)
+
+        expect(page).to_not have_content(@passenger1.name)
+        expect(page).to_not have_content(@passenger2.name)
+        expect(page).to_not have_content(@passenger3.name)
+        expect(page).to_not have_content(@passenger7.name)
+      end
+
+      within("#flight-#{@flight5.id}") do
+        expect(page).to have_content("Flight Airline: #{@airline3.name}")
+        expect(page).to have_content(@passenger2.name)
+        expect(page).to have_content(@passenger6.name)
+
+        expect(page).to_not have_content(@passenger1.name)
+        expect(page).to_not have_content(@passenger3.name)
+        expect(page).to_not have_content(@passenger4.name)
+        expect(page).to_not have_content(@passenger5.name)
+        expect(page).to_not have_content(@passenger7.name)
+      end
+
+      within("#flight-#{@flight6.id}") do
+        expect(page).to have_content("Flight Airline: #{@airline3.name}")
+        expect(page).to have_content(@passenger6.name)
+
+        expect(page).to_not have_content(@passenger1.name)
+        expect(page).to_not have_content(@passenger2.name)
+        expect(page).to_not have_content(@passenger3.name)
+        expect(page).to_not have_content(@passenger4.name)
+        expect(page).to_not have_content(@passenger5.name)
+        expect(page).to_not have_content(@passenger7.name)
+      end
+
+      within("#flight-#{@flight7.id}") do
+        expect(page).to have_content("Flight Airline: #{@airline4.name}")
+
+        expect(page).to_not have_content(@passenger1.name)
+        expect(page).to_not have_content(@passenger2.name)
+        expect(page).to_not have_content(@passenger3.name)
+        expect(page).to_not have_content(@passenger4.name)
+        expect(page).to_not have_content(@passenger5.name)
+        expect(page).to_not have_content(@passenger6.name)
+        expect(page).to_not have_content(@passenger7.name)
+      end
+
+      within("#flight-#{@flight8.id}") do
+        expect(page).to have_content("Flight Airline: #{@airline4.name}")
+
+        expect(page).to_not have_content(@passenger1.name)
+        expect(page).to_not have_content(@passenger2.name)
+        expect(page).to_not have_content(@passenger3.name)
+        expect(page).to_not have_content(@passenger4.name)
+        expect(page).to_not have_content(@passenger5.name)
+        expect(page).to_not have_content(@passenger6.name)
+        expect(page).to_not have_content(@passenger7.name)
+      end
```

Comment:
In a project, I would consider this redundant testing, but for this IC, I get it and it's fine. :) Good job using within blocks.